src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax;
+
+import core.basesyntax.model.Circle;
+import core.basesyntax.model.Figure;
+import core.basesyntax.model.IsoscelesTrapezoid;
+import core.basesyntax.model.Rectangle;
+import core.basesyntax.model.RightTriangle;
+import core.basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 4;
+    public static final int PART_COUNT = 50;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String randomColor = colorSupplier.getRandomColor();
+        int randomPart1 = random.nextInt(PART_COUNT);
+        int randomPart2 = random.nextInt(PART_COUNT);
+        int randomPart3 = random.nextInt(PART_COUNT);
+        Figure figure = null;
+        switch (figureNumber) {
+
+            case 0: {
+                figure = new Square(randomPart1, randomColor);
+                break;
+
+            }
+            case 1: {
+                figure = new Rectangle(randomPart1, randomPart2, randomColor);
+                break;
+
+            }
+            case 2: {
+                figure = new Circle(randomPart1, randomColor);
+                break;
+            }
+            case 3: {
+                figure = new RightTriangle(randomPart1, randomPart2, randomColor);
+                break;
+
+            }
+            case 4: {
+                figure = new IsoscelesTrapezoid(randomPart1, randomPart2, randomPart3, randomColor);
+                break;
+
+            }
+            default: {
+                System.out.println("Switch default");
+                break;
+            }
+
+        }
+        return figure;
```

Comment:
1. unnecessary empty lines
2. unnecessary breaks
3. returning null is a very bad practice, it's better to remove case 4 and make last figure to be returned in default case
```suggestion
            case 0: {
                return new Square(randomPart1, randomColor);
            }
            case 1: {
                return new Rectangle(randomPart1, randomPart2, randomColor);
            }
            case 2: {
                return new Circle(randomPart1, randomColor);
            }
            case 3: {
                return new RightTriangle(randomPart1, randomPart2, randomColor);
            }
            default: {
                return new IsoscelesTrapezoid(randomPart1, randomPart2, randomPart3, randomColor);
            }
        }
```